From 08b496475483d3f0f75c144191469f070e2629d4 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Thu, 29 Dec 2022 15:16:46 +0900
Subject: [PATCH] test: use hash explicitly for Struct initializer for ruby 3.2

ruby 3.2 changes Struct.new behavior so that keyword_init option is
enabled by default. This means that keyword argument passed to
the initializer of Struct is now treated differently with ruby3.2.

To avoid behavior change in jekyll test code, use hash explicitly.

Fixes #9236 .
---
 test/test_filters.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test_filters.rb b/test/test_filters.rb
index ea82fc5b5e3..ee9b31af036 100644
--- a/test/test_filters.rb
+++ b/test/test_filters.rb
@@ -718,7 +718,7 @@ def to_liquid
           {
             "name" => name,
             :v     => 1,
-            :thing => M.new(:kay => "jewelers"),
+            :thing => M.new({:kay => "jewelers"}),
             :stuff => true,
           }
         end
